workspaces/servicenow/plugins/servicenow-backend/src/service-now-rest/client.ts

```diff
@@ -303,4 +309,101 @@ export class DefaultServiceNowClient implements ServiceNowClient {
 
     throw new Error(`User with email ${email} not found in ServiceNow.`);
   }
+
+  async getBusinessApplication(appCode: string) {
+    try {
+      const apiUrl = new URL(
+        'cmdb_ci_business_app',
+        `${this.cmdbBaseUrl}api/now/table/`,
+      );
+      apiUrl.searchParams.append(
+        'sysparm_query',
+        `u_application_id=${appCode}`,
+      );
+      const response = await axios.get(apiUrl.toString(), {
+        headers: {
+          Authorization: `${this.config.servicenow?.cmdbToken}`,
+          Accept: 'application/json',
+        },
+      });
+
+      if (response.status >= 400 && response.status < 600) {
+        throw new Error('Failed to fetch application details');
+      }
+
+      return response.data;
+    } catch (error: any) {
+      this.logger.error(
+        `Failed to fetch application details: ${error.message}`,
+        {
+          error,
+        },
+      );
+      throw new Error(`Failed to fetch application details: ${error.message}`);
+    }
+  }
+
+  async getUserDetails(userId: string) {
+    try {
+      const apiUrl = new URL(
+        `sys_user/${userId}`,
+        `${this.cmdbBaseUrl}api/now/table/`,
+      );
+
+      const response = await axios.get(apiUrl.toString(), {
+        headers: {
+          Authorization: `${this.config.servicenow?.cmdbToken}`,
+          Accept: 'application/json',
+        },
+      });
+
+      if (response.status >= 400 && response.status < 600) {
+        throw new Error('Failed to fetch user details');
+      }
+
+      return response.data;
+    } catch (error: any) {
+      this.logger.error(`Failed to fetch user details: ${error.message}`, {
+        error,
+      });
+      throw new Error(`Failed to fetch user details: ${error.message}`);
+    }
+  }
+
+  async getInfraDetails(appCode: string) {
+    try {
+      const apiUrl = new URL(
+        `cmdb_rel_ci`,
+        `${this.cmdbBaseUrl}api/now/table/`,
+      );
+      apiUrl.searchParams.append(
+        'sysparm_query',
+        `parent.u_business_app.u_application_idSTARTSWITH${appCode}^type.nameSTARTSWITHHosted on::Hosts^GROUPBYchild.name`,
+      );
+      apiUrl.searchParams.append(
+        'sysparm_fields',
+        'parent.name,parent.sys_class_name,u_display,child.name,sys_updated_on',
+      );
```

Comment:
Is it worth to build this strings more dynamically?

Or do you think we should check if we can use the servicenow sdk to make this API calls? :thinking: